### ISSUE TITLE
Crash when skill has non-github remote origin url

### ISSUE THREAD

**Issue by tristanschneider2000** [MycroftAI/mycroft-skills-manager#48]
I wrote a local skill that is not on GitHub. It worked fine until I added a remote or cloned the skill from a git remote. This remote is located on my personal server and is accessed by git via ssh.
As soon as I tried restarting the skills, I noticed that not a single skill would load. When tying to use msm, I got this error:
```
(.venv) pi@mycroft:/opt/mycroft/skills $ msm search mycroft-hello-world
Traceback (most recent call last):
  File "/home/pi/mycroft-core/.venv/bin/msm", line 11, in <module>
    sys.exit(main())
  File "/home/pi/mycroft-core/.venv/lib/python3.5/site-packages/msm/__main__.py", line 97, in main
    args.platform, args.skills_dir, repo, args.versioned
  File "/home/pi/mycroft-core/.venv/lib/python3.5/site-packages/msm/mycroft_skills_manager.py", line 85, in __init__
    self.sync_skills_data()
  File "/home/pi/mycroft-core/.venv/lib/python3.5/site-packages/msm/mycroft_skills_manager.py", line 189, in sync_skills_data
    self.skills_data = self.load_skills_data()
  File "/home/pi/mycroft-core/.venv/lib/python3.5/site-packages/msm/mycroft_skills_manager.py", line 184, in load_skills_data
    skills_data = self.curate_skills_data(skills_data)
  File "/home/pi/mycroft-core/.venv/lib/python3.5/site-packages/msm/mycroft_skills_manager.py", line 144, in curate_skills_data
    local_skills = [s for s in self.list() if s.is_local]
  File "/home/pi/mycroft-core/.venv/lib/python3.5/site-packages/msm/mycroft_skills_manager.py", line 358, in list
    skill = SkillEntry.from_folder(dirname(skill_file), msm=self)
  File "/home/pi/mycroft-core/.venv/lib/python3.5/site-packages/msm/skill_entry.py", line 188, in from_folder
    return cls(None, path, cls.find_git_url(path), msm=msm)
  File "/home/pi/mycroft-core/.venv/lib/python3.5/site-packages/msm/skill_entry.py", line 129, in __init__
    self.author = self.extract_author(url) if url else ''
  File "/home/pi/mycroft-core/.venv/lib/python3.5/site-packages/msm/skill_entry.py", line 206, in extract_author
    return url.rstrip('/').split("/")[-2].split(':')[-1]
IndexError: list index out of range
```

A very similar error appeared in the skills log, though this time in the context of creating the skill.

To solve my problem, I took a look at the code and found out that skill_entry.py's function extract_author() extracts the GitHub username of the author from the url, and that this url is just the URL of the origin remote, which is not necessarily a GitHub URL. Because of that, the url it found for my skill was in an entirely different format that could not be handled by extract_author(), causing the IndexError after splitting the url and trying to access parts of the string that weren't there.

My temporary fix was to delete the .git folder in my local copy of that skill, though I guess just renaming the remote to something other than 'origin' would work as well.

I suggest checking the url format of the remote url before trying to extract info from it or simply some error handling in the extract_author() and similar functions, perhaps catching the IndexError and returning an empty string.

**Comment by krisgesling** [comment]
Hi Tristan, 

Thanks for flagging this and for the outline of what you found, it is a huge help for those looking to solve it. 

As an open source project, we are always very open and grateful for new contributions. Is this something you would be interested in trying to tackle?

**Comment by forslund** [comment]
Should be fixed by #52. Closing this, feel free to reopen if problem persists.
